aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -218,66 +243,47 @@ Tensor& add_sparse_(Tensor& self, const Tensor& other, Scalar alpha) {
 // There's actually nothing sparse specific about these implementations
 
 Tensor sub_sparse(const Tensor& self, const Tensor& other, Scalar alpha) {
+  sub_check(self, other);
   return native::add_sparse(self, other, -alpha);
 }
 
 Tensor& sub_sparse_(Tensor& self, const Tensor& other, Scalar alpha) {
+  sub_check(self, other);
   return native::add_sparse_(self, other, -alpha);
 }
 
 Tensor& sub_out_sparse(Tensor& r, const Tensor& self, const Tensor& other, Scalar alpha) {
+  sub_check(self, other);
   return at::add_out(r, self, other, -alpha);  // redispatch!
 }
 
-Tensor& add_out_dense_sparse_cpu(Tensor& r, const Tensor& dense, const SparseTensor& sparse_, Scalar value);
 
-SparseTensor& add_out_sparse_cpu(SparseTensor& r, const SparseTensor& t, const SparseTensor& src, Scalar value) {
-  if (!t.is_sparse()) {
-    return add_out_dense_sparse_cpu(r, t, src, value);
-  }
-  // TODO: This test seems a bit goofy
-  TORCH_CHECK(src.is_sparse(), "add(sparse, dense) is not supported. Use add(dense, sparse) instead.");
-  AT_ASSERT(!t.is_cuda());  // the dispatch argument
-  TORCH_CHECK(!r.is_cuda(), "add: expected 'out' to be CPU tensor, but got CUDA tensor");
-  TORCH_CHECK(!src.is_cuda(), "add: expected 'other' to be a CPU tensor, but got a CUDA tensor");
+SparseTensor& add_out_sparse_contiguous(SparseTensor& r, const SparseTensor& t, const SparseTensor& src, Scalar value, ScalarType commonDtype) {
```

Comment:
Is it a convention to call the promoted type the "commonDtype"? I feel like a better name is "promotedDtype" because the dtype isn't really "common" to the inputs; the inputs can be different dtypes.

Reply:
We call it common_dtype in tensoriterator so I think this makes sense. In an earlier version of this PR I called it promoted_type, and there was some objection:

https://github.com/pytorch/pytorch/pull/26051#discussion_r327831374

Reply:
Sounds reasonable